### PR TITLE
Migrate to setuptools

### PR DESCRIPTION
**Pull request by BrunoCoimbra**
This pull request updates the repository structure to support setuptools.

### CODE REVIEW

**Comment by jcpunk** [comment]
Looks like `.github/workflows/rpm-build.yaml` will also need a tweak

**Comment by jcpunk** [comment]
@StevenCTimm looks like the billing calculator in https://github.com/HEPCloud/contrib has a different name (and different actual code) from the one hiding out in pypi test.  The version string also appears to be much older.

**Comment by jcpunk** [comment]
@vitodb any chance you can look at the docker bits here.  I don't recall what we did exactly for the framework.....

**Comment by StevenCTimm** [comment]
Yes that's right--will ask Maria to check that out.
the one in pypi test is the correct one.

Steve

________________________________

**Comment by vitodb** [comment]
By looking at the failure it looks like the [entrypoint script](https://github.com/BrunoCoimbra/decisionengine_modules/blob/master/package/ci/python3-entrypoint.sh#L35)  is missing the requirement file decisionengine_modules/requirements.txt.
The action used to build the RPM is quite different in decisionengine than in decisionengine_modules, I have to dig more on this.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -254,7 +254,7 @@ jobs:
           python3 -m pip install --upgrade pip
           python3 -m pip install --upgrade pytest
           python3 -m pip install -r requirements.txt
```
Comment (jcpunk) [inline .github/workflows/ci.yaml]: This line should probably change to something like

`python3 -m pip install -e . --user`

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -254,7 +254,7 @@ jobs:
           python3 -m pip install --upgrade pip
           python3 -m pip install --upgrade pytest
           python3 -m pip install -e . --user
-          python3 -m pip install --index-url https://test.pypi.org/simple --no-deps bill-calculator-hep-mapsacosta==0.0.9
+          python3 -m pip install --no-deps bill-calculator-hep==0.0.10
```
Comment (jcpunk) [inline .github/workflows/ci.yaml:257]: I don't think this line is needed since it was installed on line 90

File: package/rpm/install_section [hunk 3]
```diff
@@ -0,0 +1,5 @@
+python3 setup.py install --single-version-externally-managed -O1 --root=%{buildroot} --record=INSTALLED_FILES
+
+# make sure we also own the top level and egg-info raw dir to make things easy on uninstall
+echo "%{python3_sitelib}/%{name}-*egg-info" >> INSTALLED_FILES
+echo "%{python3_sitelib}/%{name}" >> INSTALLED_FILES
```
Comment (jcpunk) [inline package/rpm/install_section:5]: Can you add the missing end line marker?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    "bill-calculator-hep >= 0.1.2",
+    "numpy == 1.19.5; python_version <= '3.6'",
+    "numpy >= 1.19.5; python_version >= '3.7'",
+    "pandas == 1.1.5; python_version <= '3.6'",
+    "pandas >= 1.1.5; python_version >= '3.7'"
+]
+
+# pull in development requirements
+devel_req = [
+    "mock >= 4.0.3", "pytest >= 6.2.2", "pytest-cov >= 2.11.1",
```
Comment (jcpunk) [inline setup.py:36]: I believe the `mock` module was a python2 backport of `unittest.mock` from Python3.

**Review by jcpunk — APPROVED** [review]

File: package/rpm/install_section (hunk 3 above)
Comment (BrunoCoimbra) [reply package/rpm/install_section]: I added the missing end line.

File: setup.py (hunk 4 above)
Comment (BrunoCoimbra) [reply setup.py]: I replaced the `mock` module with `unittest.mock`.

**Review by jcpunk — APPROVED** [review]

**Review by mambelli — APPROVED** [review]

**Review by jcpunk — APPROVED** [review]

**Pull request merged**